Source/DafnyCore/Verifier/BoogieGenerator.cs

```diff
@@ -3469,15 +3470,24 @@ public ForceCheckToken(IToken tok)
       return req;
     }
 
-    Bpl.StmtList TrStmt2StmtList(BoogieStmtListBuilder builder, Statement block, List<Variable> locals, ExpressionTranslator etran) {
+    Bpl.StmtList TrStmt2StmtList(BoogieStmtListBuilder builder,
+      Statement block, List<Variable> locals, ExpressionTranslator etran, bool introduceScope = false) {
       Contract.Requires(builder != null);
       Contract.Requires(block != null);
       Contract.Requires(locals != null);
       Contract.Requires(etran != null);
       Contract.Requires(codeContext != null && predef != null);
       Contract.Ensures(Contract.Result<Bpl.StmtList>() != null);
 
+      if (introduceScope) {
```

Comment:
Could you please give me some context about what this new contextual boolean is doing? e.g. where in the code this would be triggered.

Reply:
It's currently only called for translating the then and else block of an if statement, but this method is only called three times to begin with.